src/index.ts

```diff
@@ -956,36 +973,40 @@ export class ValidationService {
             this.trackFormInput(input.form, uid);
         }
 
-        if (this.elementEvents[uid]) {
+        if (this.inputEvents[uid]) {
             return;
         }
 
-        let delay;
-        let cb = e => {
+        let debounceTimeoutID = 0;
+        let cb = (e: Event, callback?: ValidatedCallback) => {
             let validate = this.validators[uid];
-            clearTimeout(delay);
-            delay = setTimeout(validate, this.debounce);
+            clearTimeout(debounceTimeoutID);
+            debounceTimeoutID = setTimeout(() => {
+                validate()
+                    .then(callback)
+                    .catch(error => {
+                        this.logger.log('Validation error', error);
+                    });
+            }, this.debounce);
         };
 
-        let isDropdown = input.tagName.toLowerCase() === 'select';
         let validateEvent = input.dataset.valEvent;
-        if (isDropdown) {
-            input.addEventListener('change', cb);
-        } else if (validateEvent) {
```

Comment:
While I'm in the area I noticed the `data-val-event` override is being ignored for `<select>`. The attribute now takes precedence over the default `change` event.